src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.MetadataDb.cs

```diff
@@ -148,7 +148,8 @@ internal MetadataDb(MetadataDb source, bool useArrayPools)
 
             public void Dispose()
             {
-                if (_data == null)
+                byte[] data = Threading.Interlocked.Exchange(ref _data, null);
```

Comment:
Please add a `using System.Threading;` at the top and then remove the `Threading.` from here and other such uses.